lib/python/rose/apps/rose_ana.py

```diff
@@ -293,58 +381,114 @@ def run(self, app_runner, conf_tree, opts, args, uuid, work_files):
-                    self.reporter(msg, prefix=self._prefix_fail)
-
-            except Exception:
-                # If an exception was raised, print a traceback and treat it
-                # as a failure.
-                task_error = True
-                number_of_failures += 1
-                msg = "Task #{0} encountered an error".format(itask + 1)
-                summary_status.append(("{0} ({1})".format(
-                    msg, task.options["full_task_name"]),
-                    self._prefix_fail))
-                self.reporter(msg + " (see stderr)", prefix=self._prefix_fail)
-                exception = traceback.format_exc()
-                self.reporter(msg, prefix=self._prefix_fail,
-                              kind=self.reporter.KIND_ERR)
-                self.reporter(exception, prefix=self._prefix_fail,
-                              kind=self.reporter.KIND_ERR)
+                    self.reporter(msg + " (see stderr)",
+                                  prefix=self._prefix_fail)
+                    exception = traceback.format_exc()
+                    self.reporter(msg, prefix=self._prefix_fail,
+                                  kind=self.reporter.KIND_ERR)
+                    self.reporter(exception, prefix=self._prefix_fail,
+                                  kind=self.reporter.KIND_ERR)
+
+        elif n_threads > 1:
+            self.reporter("Running in THREADED mode, with {0} threads"
+                          .format(n_threads))
+            # Multithreaded case
+            # Create threading objects for each comparison task
+            threads = []
+            for itask, task in enumerate(self.analysis_tasks):
+                threads.append(TaskRunner(self, itask, task))
+
+            # Start threads within the set number of concurrent threads until
+            # all have been started
+            itask = 0
+            while itask < len(threads):
+                if threading.active_count() < n_threads:
+                    self.reporter(
+                        "Starting thread for task {0} at {1}"
+                        .format(itask + 1, timestamp()))
+                    threads[itask].start()
+                    itask += 1
+
+            # Gather up the results
+            number_of_failures = 0
+            number_of_skips = 0
+            task_error = False
+            summary_status = []
+            for thread in threads:
+                # If any threads haven't finished yet make sure to wait
+                thread.join()
+                number_of_failures += thread.failures
+                number_of_skips += thread.skips
+                task_error = (task_error or thread.task_error)
+                summary_status += thread.summary_status
+                # And print the output via the main reporter
+                for args in thread.reporter_args:
+                    self.reporter(*args)
+
+        else:
+            # Negative threads?
```

Comment:
Would
```python
msg = (f"Number of threads must be a positive integer, "
       f"so should not be {n_threads}")
```
cover more cases, and offer more assistance to the end user? 

Could you add tests for n_threads = [-1, 0, 0.25, "Tim"]) to check this loop?